lib/rex/post/meterpreter/packet_dispatcher.rb

```diff
@@ -393,11 +390,16 @@ def monitor_socket
           ::IO.select(nil, nil, nil, 0.10)
         end
 
-        @pqueue.unshift(*incomplete)
+        while incomplete.length > 0
+          @pqueue << incomplete.shift
+        end
 
         if(@pqueue.length > 100)
-          dlog("Backlog has grown to over 100 in monitor_socket, dropping older packets: #{@pqueue[0 .. 25].map{|x| x.inspect}.join(" - ")}", 'meterpreter', LEV_1)
-          @pqueue = @pqueue[25 .. 100]
+          removed = []
+          (1..25).each {
+            removed << @pqueue.pop
```

Comment:
This should be a `shift`. `Queue#pop` takes the most recently added, so you're killing new packets, not older.